engine/class_modules/sc_druid.cpp

```diff
@@ -4509,27 +4704,92 @@ struct natures_guardian_t : public druid_heal_t
   }
 };
 
+// Nature's Swiftness =======================================================
+struct natures_swiftness_t : public druid_heal_t
+{
+  natures_swiftness_t( druid_t* p, std::string_view opt )
+    : druid_heal_t( "natures_swiftness", p, p->spec.natures_swiftness, opt )
+  {}
+
+  timespan_t cooldown_duration() const override
+  {
+    return 0_ms;  // cooldown handled by buff.natures_swiftness
+  }
+
+  void execute() override
+  {
+    druid_heal_t::execute();
+
+    p()->buff.natures_swiftness->trigger();
+  }
+
+  bool ready() override
+  {
+    if ( p()->buff.natures_swiftness->check() )
+      return false;
+
+    return druid_heal_t::ready();
+  }
+};
+
+// Nourish ==================================================================
+struct nourish_t : public druid_heal_t
+{
+  nourish_t( druid_t* p, std::string_view opt ) : druid_heal_t( "nourish", p, p->talent.nourish ) {}
+
+  double harmony_multiplier( player_t* t ) const override
+  {
+    return druid_heal_t::harmony_multiplier( t ) * 3.0;
+  }
+};
+
 // Regrowth =================================================================
 struct regrowth_t : public druid_heal_t
 {
-  timespan_t gcd_add;
-
-  regrowth_t( druid_t* p, std::string_view options_str )
-    : druid_heal_t( "regrowth", p, p->find_class_spell( "Regrowth" ), options_str )
+  struct regrowth_state_t : public action_state_t
   {
-    form_mask             = NO_FORM | MOONKIN_FORM;
-    may_autounshift       = true;
-    ignore_false_positive = true;  // Prevents cat/bear from failing a skill check and going into caster form.
+    double bonus_crit;
 
-    // Hack for feral to be able to use target-related expressions on this action
-    // Disables healing entirely
-    if ( p->specialization() == DRUID_FERAL )
+    regrowth_state_t( action_t* a, player_t* t ) : action_state_t( a, t ), bonus_crit( 0.0 ) {}
+
+    void initialize() override
     {
-      target          = sim->target;
-      base_multiplier = 0;
+      action_state_t::initialize();
+
+      bonus_crit = 0.0;
     }
 
-    gcd_add = p->query_aura_effect( p->spec.cat_form, A_ADD_FLAT_MODIFIER, P_GCD, s_data )->time_value();
+    void copy_state( const action_state_t* s ) override
+    {
+      action_state_t::copy_state( s );
+
+      bonus_crit = debug_cast<const regrowth_state_t*>( s )->bonus_crit;
+    }
+
+    double composite_crit_chance() const override
+    {
+      return action_state_t::composite_crit_chance() + bonus_crit;
+    }
+  };
+
+  timespan_t gcd_add;
+  double bonus_crit;
+  double sotf_mul;
+
+  regrowth_t( druid_t* p, std::string_view opt )
+    : druid_heal_t( "regrowth", p, p->find_class_spell( "Regrowth" ), opt ),
+      sotf_mul( p->buff.soul_of_the_forest_tree->data().effectN( 1 ).percent() )
+  {
+    form_mask = NO_FORM | MOONKIN_FORM;
+    may_autounshift = true;
+
+    gcd_add = p->query_aura_effect( p->spec.cat_form, A_ADD_FLAT_MODIFIER, P_GCD, &data() )->time_value();
```

Comment:
you could set these 2 variables in the initializer list of the constructor, if you want.